pulsar-broker/src/main/java/org/apache/pulsar/broker/loadbalance/impl/LinuxBrokerHostUsageImpl.java

```diff
@@ -252,10 +252,18 @@ private double getTotalNicLimitKbps(List<String> nics) {
                 .orElseGet(() -> nics.stream().mapToDouble(nicPath -> {
                     // Nic speed is in Mbits/s, return kbits/s
                     try {
-                        return Double.parseDouble(new String(Files.readAllBytes(getNicSpeedPath(nicPath))));
+                        final Path nicSpeedPath = getNicSpeedPath(nicPath);
+                        return readDoubleFromPath(nicSpeedPath);
                     } catch (IOException e) {
-                        log.error(String.format("Failed to read speed for nic %s, maybe you can set broker"
-                                + " config [loadBalancerOverrideBrokerNicSpeedGbps] to override it.", nicPath), e);
+                        if ("Invalid argument".equals(e.getMessage())) {
```

Comment:
we should add some comments here and add references to when/why/where this could happen

Reply:
I added the more info I could. I'm not able to find Linux docs to explain the message matching but it seems a standard. I looked at the Linux source code and I haven't found any other error messages that can be raised in this context.